docs/graphql.md

```diff
@@ -147,6 +147,77 @@ during your boot process.
 The goal is to have all clients created at boot so there's no schema loading,
 parsing, or client instantiation done during runtime when your app serves a request.
 
+
+## Using a custom GraphQL Client
+By default `ShopifyAPI::GraphQL` wraps the Github GraphQL Client library. However, this client
+may not suitable for various reasons. If you wish to expand on the interface of the client or
+improve the required functions for your use case you can implement a client of your own.
+
+The client must conform to the following interface:
```

Comment:
I'm a bit wary of detailing this interface in docs since it could change. I'm not sure it's reasonable (or common) for people to implement their own client entirely that conforms to this, maybe we should just mention/expect people to subclass `::GraphQL::Client`? If someone really wants to go beyond that then they're still free to.

Reply:
I'd hope that if someone has gone to this extent that they wouldn't need the doc, but also know to inherit from `GraphQL::Client`, totally agree. I was going with the mindset that it was easier to do too much and remove it.